includes/discovery/sensors/state/timos.inc.php

```diff
@@ -0,0 +1,79 @@
+
+$nokia_device_state_name  = 'tmnxDeviceState';
+$nokia_device_state_index = create_state_index($nokia_device_state_name);
+if ($nokia_device_state_index !== null) {
+    $nokia_device_states      = array(
+        array($nokia_device_state_index, 'Unknown', 1,1,3),
+        array($nokia_device_state_index, 'Not Equippped', 1,2,3),
+        array($nokia_device_state_index, 'Ok', 1,3,0),
+        array($nokia_device_state_index, 'Failed', 1,4,2),
+        array($nokia_device_state_index, 'Out Of Service', 1,5,1),
+        array($nokia_device_state_index, 'Not Provisioned', 1,6,3),
+    );
+    foreach ($nokia_device_states as $state) {
+        $insert = array(
+            'state_index_id'        => $state[0],
+            'state_descr'           => $state[1],
+            'state_draw_graph'      => $state[2],
+            'state_value'           => $state[3],
+            'state_generic_value'   => $state[4],
+        );
+        dbInsert($insert, 'state_translations');
+    }
+}
+
+$nokia_device_ps_assigned_types_name  = 'tmnxChassisPowerSupplyAssignedTypes';
+$nokia_device_ps_assigned_types_index = create_state_index($nokia_device_ps_assigned_types_name);
+if ($nokia_device_ps_assigned_types_index !== null) {
+    $nokia_device_ps_assigned_types      = array(
+        array($nokia_device_ps_assigned_types_index, 'None', 0,0,0),
+        array($nokia_device_ps_assigned_types_index, 'DC', 0,1,0),
+        array($nokia_device_ps_assigned_types_index, 'Single AC', 0,2,0),
+        array($nokia_device_ps_assigned_types_index, 'Multiple AC', 0,3,0),
+        array($nokia_device_ps_assigned_types_index, 'default', 0,4,3),
+    );
+    foreach ($nokia_device_ps_assigned_types as $state) {
+        $insert = array(
+          'state_index_id'        => $state[0],
+          'state_descr'           => $state[1],
+          'state_draw_graph'      => $state[2],
+          'state_value'           => $state[3],
+          'state_generic_value'   => $state[4],
+        );
+        dbInsert($insert, 'state_translations');
+    }
+}
+
+$base_oid          = '.1.3.6.1.4.1.6527.3.1.2.2.1.5.1';
+$power_supply_oids = array(
+    array('sub_oid' => '2.1.1', 'desc' => 'AC Status', 'type' => $nokia_device_state_name),
+    array('sub_oid' => '3.1.1', 'desc' => 'DC Status', 'type' => $nokia_device_state_name),
+    array('sub_oid' => '6.1.1', 'desc' => 'Power Supply 1', 'type' => $nokia_device_state_name),
+    array('sub_oid' => '7.1.1', 'desc' => 'Power Supply 2', 'type' => $nokia_device_state_name),
+    array('sub_oid' => '8.1.1', 'desc' => 'Power Supply Type', 'type' => $nokia_device_ps_assigned_types_name),
+    array('sub_oid' => '9.1.1', 'desc' => 'Power In', 'type' => $nokia_device_state_name),
+    array('sub_oid' => '10.1.1', 'desc' => 'Power Out', 'type' => $nokia_device_state_name),
+);
+
+foreach ($power_supply_oids as $data) {
+    $full_oid = $base_oid . "." . $data['sub_oid'];
```

Comment:
You should just add the trailing `.` to `$base_oid` to make this easier to read.